sonar-dotnet-shared-library/src/main/java/org/sonar/plugins/dotnet/tests/ScannerFileService.java

```diff
@@ -51,39 +50,29 @@ public boolean isSupportedAbsolute(String absolutePath) {
   }
 
   public Optional<InputFile> getFileByRelativePath(String filePath) {
-    String normalizedRelativePath = getNormalizedRelativePath(filePath);
-    Iterable<InputFile> files = fileSystem.inputFiles(fileSystem.predicates().hasLanguage(languageKey));
-    int count = 0;
-    InputFile foundFile = null;
-    for (InputFile file : files) {
-      String path = file.uri().getPath();
-      if (path.endsWith(normalizedRelativePath)) {
-        count++;
-        foundFile = file;
-      }
-    }
-    if (count == 1) {
-      LOG.trace("Found indexed file '{}' for '{}' (normalized to '{}').", foundFile.uri().getPath(), filePath, normalizedRelativePath);
-      return Optional.of(foundFile);
-    } else {
-      LOG.debug("Found {} indexed files for '{}' (normalized to '{}'). Will skip this coverage entry. Verify sonar.sources in .sonarqube\\out\\sonar-project.properties.", count, filePath, normalizedRelativePath);
-      return Optional.empty();
-    }
-  }
-
-  private static String getNormalizedRelativePath(String filePath) {
-    String relativePath = replaceDeterministicSourcePath(filePath);
-    return relativePath.replace('\\', '/');
-  }
-
-  private static String replaceDeterministicSourcePath(String filePath) {
-    Matcher matcher = DETERMINISTIC_SOURCE_PATH_PREFIX.matcher(filePath);
-    if (matcher.find())
-    {
+    Matcher matcher = DETERMINISTIC_SOURCE_PATH_PREFIX.matcher(filePath.replace('\\', '/'));
+    if (matcher.find()) {
       String relativePath = matcher.replaceFirst("");
-      LOG.trace("It seems Deterministic Source Paths are used, replacing '{}' with '{}'.", filePath, relativePath);
-      return relativePath;
+      Iterable<InputFile> files = fileSystem.inputFiles(fileSystem.predicates().hasLanguage(languageKey));
+      int count = 0;
+      InputFile foundFile = null;
+      for (InputFile file : files) {
+        String path = file.uri().getPath();
+        if (path.endsWith(relativePath)) {
+          count++;
+          foundFile = file;
+        }
+      }
+      if (count == 1) {
+        LOG.trace("Found indexed file '{}' for '{}' (normalized to '{}').", foundFile.uri().getPath(), filePath, relativePath);
+        return Optional.of(foundFile);
+      } else {
+        LOG.debug("Found {} indexed files for '{}' (normalized to '{}'). Will skip this coverage entry. Verify sonar.sources in .sonarqube\\out\\sonar-project.properties.",
```

Comment:
`.sonarqube\\out\\sonar-project.properties` should use the local system path separator - I'll make a separate PR for that, as I want to improve the general logging